rpcs3/Emu/Cell/Modules/cellGem.cpp

```diff
@@ -299,6 +285,112 @@ static bool map_ext_to_ds3_input(const u32 port_no, CellGemExtPortData& ext)
 	return true;
 }
 
+/**
+ * \brief Maps Move controller data (digital buttons, and analog Trigger data) to mouse input.
+ *        Move Button: Mouse1
+ *        Trigger:     Mouse2
+ * \param mouse_no Mouse index number to use
+ * \param digital_buttons Bitmask filled with CELL_GEM_CTRL_* values
+ * \param analog_t Analog value of Move's Trigger.
+ * \return true on success, false if mouse mouse_no is invalid
+ */
+static bool mouse_input_to_pad(const u32 mouse_no, be_t<u16>& digital_buttons, be_t<u16>& analog_t)
+{
+	const auto handler = g_fxo->get<MouseHandlerBase>();
+
+	std::lock_guard lock(handler->mutex);
+
+	if (!handler || mouse_no >= handler->GetMice().size())
+	{
+		return false;
+	}
+
+	memset(&digital_buttons, 0, sizeof(digital_buttons));
```

Comment:
This needs a temp var.
```suggestion
	const auto& mouse_data = handler->GetMice().at(0);

	u16 buttons_data = 0;
	if (mouse_data.buttons & CELL_MOUSE_BUTTON_1)
		buttons_data |= CELL_GEM_CTRL_T;
	if (mouse_data.buttons & CELL_MOUSE_BUTTON_2)
		buttons_data |= CELL_GEM_CTRL_MOVE;
	if (mouse_data.buttons & CELL_MOUSE_BUTTON_3)
		buttons_data |= CELL_GEM_CTRL_CROSS;
	if (mouse_data.buttons & CELL_MOUSE_BUTTON_3 && mouse_data.buttons & CELL_MOUSE_BUTTON_2)
		buttons_data |= CELL_GEM_CTRL_TRIANGLE;
	if (mouse_data.buttons & CELL_MOUSE_BUTTON_3 && mouse_data.buttons & CELL_MOUSE_BUTTON_1)
		buttons_data |= CELL_GEM_CTRL_START;
	digital_buttoms = buttons_data;
```